packages/common/container-definitions/src/loader.ts

```diff
@@ -102,19 +102,128 @@ export interface ICodeAllowList {
+     *
+     * @see {@link IContainer.readOnlyInfo}
+     */
     (event: "readonly", listener: (readonly: boolean) => void): void;
+
+    /**
+     * Emitted when the {@link IContainer} completes connecting to the Fluid service.
+     *
+     * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+     *
+     * @see
+     *
+     * - {@link IContainer.connectionState}
+     *
+     * - {@link IContainer.connect}
+     */
     (event: "connected", listener: (clientId: string) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `codeDetails`: TODO
+     *
+     * - `proposal`: TODO
+     *
+     * @see {@link IContainer.proposeCodeDetails}
+     */
     (event: "codeDetailsProposed", listener: (codeDetails: IFluidCodeDetails, proposal: ISequencedProposal) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `codeDetails`: TODO
+     */
     (event: "contextChanged", listener: (codeDetails: IFluidCodeDetails) => void);
-    (event: "disconnected" | "attached", listener: () => void);
+
+    /**
+     * Emitted when the {@link IContainer} becomes disconnected from the Fluid service.
+     *
+     * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+     *
+     * @see
+     *
+     * - {@link IContainer.connectionState}
+     *
+     * - {@link IContainer.disconnect}
+     */
+    (event: "disconnected", listener: () => void);
+
+    /**
+     * TODO
+     */
+    (event: "attached", listener: () => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `error`: TODO
+     */
     (event: "closed", listener: (error?: ICriticalContainerError) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `error`: TODO
+     */
     (event: "warning", listener: (error: ContainerWarning) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `message`: TODO
+     */
     (event: "op", listener: (message: ISequencedDocumentMessage) => void);
-    (event: "dirty" | "saved", listener: (dirty: boolean) => void);
+
+    /**
+     * Emitted when the first local change has been made, following a "saved" event.
+     *
+     * @remarks
+     *
+     * The "saved" event will be emitted once all local changes have been acknowledged by the service.
+     *
+     * TODO: what does listener param represent in this case?
+     *
+     * @see {@link IContainer.isDirty}
+     */
+    (event: "dirty", listener: (dirty: boolean) => void);
+
+    /**
+     * Emitted when all local changes/edits have been acknowledged by the service.
+     *
+     * @remarks
+     *
+     * The "dirty" event will be emitted when the next local change has been made.
+     *
+     * TODO: what does listener param represent in this case?
```

Comment:
It doesn't appear as though the "dirty" parameter in the callback here is ever used, at least not in our codebase. What is it intended to mean? Is it not redundant with the event invocation itself?

Reply:
This just seems wrong. Please check Container class. I do not think we send this parameter at all.

Reply:
IFluidContainer does not have a flag on its events. Every usage of this and the "dirty" event in our repo ignore the flag.

Reply:
@ChumpChief noted that this is legacy and should not be recommended for use. I will add a comment indicating this. Should we remove this flag on `next`?